default/methods/source_control/file/file_browser.py

```diff
@@ -372,6 +372,54 @@ def get_instance_buffer(session, video_file_id, start, end):
     return buffer
 
 
+@routes.route('/api/project/<string:project_string_id>/files', methods = ['GET'])
```

Comment:
This route is mostly a copy of the original route `/file/list` below, with a few tweaks (discussed in comments)

The route is now `GET /project/<project_id>/files` and accepts a query parameter called `query`. This feels more natural, since we are fetching a list of files and can optionally provide a query string to filter the files